renpy/common/00action_control.rpy

```diff
@@ -201,16 +202,24 @@ init -1500 python:
 
         _layer = None
 
-        def __init__(self, screen, transition=None, _layer=None):
+        def __init__(self, screen=None, transition=None, _layer=None):
             self.screen = screen
             self.transition = transition
             self._layer = _layer
 
         def __call__(self):
+            if self.screen is None:
+                cs = renpy.current_screen()
+
+                if cs is None:
+                    return
+
+                self.screen = cs.screen_name
```

Comment:
In order to work with screens that declare a preferred `layer`, and also those called with `_layer=`, I think we also need to set the layer from the current screen before calling `hide_screen`.
```suggestion
                self.screen = cs.screen_name
                self._layer = cs.layer
```

Reply:
Well spotted!